src/components/dashboard/new_widget_popup/color_controller_ui.gd

```diff
@@ -33,8 +35,13 @@ func _on_condition_changed():
 
 func set_color_controller(controller : ColorController):
 	color_controller = controller
+	controller.connect("tree_exiting", self, "_on_color_controller_tree_exiting")
```

Comment:
I didn't test the PR, but are you sure that you want to use `tree_exiting` and not `tree_exited`?